src/extremeweatherbench/utils.py

```diff
@@ -303,3 +298,96 @@ def process_dataarray_for_output(da_list: List[Optional[xr.DataArray]]):
         output_da = output_da.isel(lead_time=slice(None, None, -1))
     output_da = expand_lead_times_to_6_hourly(output_da)
     return output_da
+
+
+def center_forecast_on_time(da: xr.DataArray, time: pd.Timestamp, hours: int):
+    """Center a forecast DataArray on a given time with a given range in hours.
+
+    Args:
+        da: The forecast DataArray to center.
+        time: The time to center the forecast on.
+        hours: The number of hours to include in the centered forecast.
+    """
+    time_range = pd.date_range(
+        end=pd.to_datetime(time) + pd.Timedelta(hours=hours),
```

Comment:
I was surprised this works :) nifty trick to create a date range ending on a precise timestamp. Cool!